pkg/coastguard/controller/controller.go

```diff
@@ -0,0 +1,58 @@
+package controller
+
+import (
+	"sync"
+
+	"github.com/submariner-io/submariner/pkg/federate"
+	"k8s.io/klog"
+
+	"github.com/submariner-io/coastguard/pkg/remotecluster"
+)
+
+// arbitrary event channel size
+const eventChannelSize = 1000
+
+type CoastguardController struct {
+	federate.ClusterEventHandler
+
+	// remoteClusters is a map of remote clusters, which have been discovered
+	remoteClusters map[string]*remotecluster.RemoteCluster
+	// syncedClusters is a map of remote clusters, which have been discovered,
+	// and also our local cache is in sync with them
+	syncedClusters map[string]*remotecluster.RemoteCluster
+
+	clusterEvents chan remotecluster.Event
+
+	processingMutex *sync.Mutex
```

Comment:
What do yoy mean exactly?

Reply:
I mean you could declare it as an unnamed field, ie:

`sync.Mutex`

Kind of like inheritance so you can just do `Lock` instead of `processingMutex.Lock`, also you don't have to initialize the field. But it's fine either way.